x/autopilot/keeper/redeem_stake_test.go

```diff
@@ -0,0 +1,253 @@
+
+	addr1 := sdk.AccAddress(ed25519.GenPrivKey().PubKey().Address().Bytes())
+	testCases := []struct {
+		forwardingActive bool
+		recvDenom        string
+		packetData       transfertypes.FungibleTokenPacketData
+		expSuccess       bool
+		expRedeemStake   bool
+	}{
+		{ // params not enabled
+			forwardingActive: false,
+			packetData: transfertypes.FungibleTokenPacketData{
+				Denom:    stAtomFullDenomPath,
+				Amount:   "1000000",
+				Sender:   "cosmos16plylpsgxechajltx9yeseqexzdzut9g8vla4k",
+				Receiver: getRedeemStakeStakeibcPacketMetadata(addr1.String(), "cosmos16plylpsgxechajltx9yeseqexzdzut9g8vla4k", ""),
+				Memo:     "",
+			},
+			recvDenom:      stAtomDenom,
+			expSuccess:     false,
+			expRedeemStake: false,
+		},
+		{ // strd denom
+			forwardingActive: true,
+			packetData: transfertypes.FungibleTokenPacketData{
+				Denom:    strdIbcDenom,
+				Amount:   "1000000",
+				Sender:   "cosmos16plylpsgxechajltx9yeseqexzdzut9g8vla4k",
+				Receiver: getRedeemStakeStakeibcPacketMetadata(addr1.String(), "cosmos16plylpsgxechajltx9yeseqexzdzut9g8vla4k", ""),
+				Memo:     "",
+			},
+			recvDenom:      "ustrd",
+			expSuccess:     false,
+			expRedeemStake: false,
+		},
+		{ // all okay
+			forwardingActive: true,
+			packetData: transfertypes.FungibleTokenPacketData{
+				Denom:    stAtomFullDenomPath,
+				Amount:   "1000000",
+				Sender:   "cosmos16plylpsgxechajltx9yeseqexzdzut9g8vla4k",
+				Receiver: getRedeemStakeStakeibcPacketMetadata(addr1.String(), "cosmos16plylpsgxechajltx9yeseqexzdzut9g8vla4k", ""),
+				Memo:     "",
+			},
+			recvDenom:      stAtomDenom,
+			expSuccess:     true,
+			expRedeemStake: true,
+		},
+		{ // all okay with memo liquidstaking since ibc-go v5.1.0
+			forwardingActive: true,
+			packetData: transfertypes.FungibleTokenPacketData{
+				Denom:    stAtomFullDenomPath,
+				Amount:   "1000000",
+				Sender:   "cosmos16plylpsgxechajltx9yeseqexzdzut9g8vla4k",
+				Receiver: addr1.String(),
+				Memo:     getRedeemStakeStakeibcPacketMetadata(addr1.String(), "cosmos16plylpsgxechajltx9yeseqexzdzut9g8vla4k", ""),
+			},
+			recvDenom:      stAtomDenom,
+			expSuccess:     true,
+			expRedeemStake: true,
+		},
+		{ // invalid receiver
+			forwardingActive: true,
+			packetData: transfertypes.FungibleTokenPacketData{
+				Denom:    stAtomFullDenomPath,
+				Amount:   "1000000",
+				Sender:   "cosmos16plylpsgxechajltx9yeseqexzdzut9g8vla4k",
+				Receiver: getRedeemStakeStakeibcPacketMetadata("xxx", "cosmos16plylpsgxechajltx9yeseqexzdzut9g8vla4k", ""),
+				Memo:     "",
+			},
+			recvDenom:      stAtomDenom,
+			expSuccess:     false,
+			expRedeemStake: false,
+		},
+		{ // invalid redeem receiver
+			forwardingActive: true,
+			packetData: transfertypes.FungibleTokenPacketData{
+				Denom:    stAtomFullDenomPath,
+				Amount:   "1000000",
+				Sender:   "cosmos16plylpsgxechajltx9yeseqexzdzut9g8vla4k",
+				Receiver: getRedeemStakeStakeibcPacketMetadata(addr1.String(), "xxx", ""),
```

Comment:
nit: do you think it's worth making the "xxx" here the properly formatted address for another chain (e.g. `osmo16plylpsgxechajltx9yeseqexzdzut9g0hvdry`)?

Reply:
good call! I'm going to add a test case to the unit test branch

Reply:
added [here](https://github.com/Stride-Labs/stride/pull/1030/commits/d71a9407501c9994cf34dbabaf12f3ceb4e630ce)!